### PR TITLE
sys/shell/commands/sc_lwip_netif.c: add address state

### PR DESCRIPTION
**Pull request by maribu**
### Contribution description

Add information about the address state (e.g. if it is valid, preferred, tentative, how many probes have been sent if it is still tentative). This is helpful for debugging an issue with IPv6 when netifs require some time to establish a link (e.g. the STM32 Ethernet driver only reports the link to be up once the auto-negotation is completed if the `stm32_eth_auto` is used).

### Testing procedure

Type `ifconfig` in an app that uses the shell, `shell_commands`, `lwip_netif` and IPv6. 

### Issues/PRs references

None

### CODE REVIEW

**Review by benpicco — APPROVED** [review]

File: sys/shell/commands/sc_lwip_netif.c [hunk 1]
```diff
@@ -75,8 +91,9 @@ static void _netif_list(struct netif *netif) {
 
 #ifdef MODULE_LWIP_IPV6
     for (i = 0; i < LWIP_IPV6_NUM_ADDRESSES; i++) {
-        if (netif_ip6_addr_state(netif, i) != IP6_ADDR_INVALID) {
-            _netif_list_ipv6(netif, i);
+        uint8_t state = netif_ip6_addr_state(netif, i);
+        if (state != IP6_ADDR_INVALID) {
```
Comment (yarrick) [inline sys/shell/commands/sc_lwip_netif.c:95]: We could do `!ip6_addr_isinvalid(state)` here to use the helpers everywhere. Either way is fine.

Comment (maribu) [reply sys/shell/commands/sc_lwip_netif.c:95]: 👍, will do

Comment (maribu) [reply sys/shell/commands/sc_lwip_netif.c]: Done. I also added a comment to point out that `ip6_addr_isvalid()` and `!ip6_addr_isinvalid()` are not the same.

**Comment by maribu** [comment]
Thanks for the reviews!

**Pull request merged**